cpp-client/deephaven/client/src/client.cc

```diff
@@ -61,7 +61,14 @@ Client::Client(std::shared_ptr<impl::ClientImpl> impl) : impl_(std::move(impl))
 }
 Client::Client(Client &&other) noexcept = default;
 Client &Client::operator=(Client &&other) noexcept = default;
-Client::~Client() = default;
+
+// There is only one Client associated with the server connection. Clients can only be moved, not copied.
+// When the client owning the state is destructed, we tear down the connection.
+Client::~Client() {
+  if (impl_ != nullptr) {
```

Comment:
I think you can do `if (impl_)` and that's a regular idiom?

Reply:
Given the number of times C's true-ish tests / implicit conversions have almost destroyed humanity, I personally err on having a coding style that is more explicit.

That said, if you feel strongly, I'll change it.

Reply:
No, don't feel strongly, keep it.